packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -14,10 +20,45 @@ declare global {
  * A styled cell that is used within the nimble-table-row.
  * @internal
  */
-export class TableCell extends FoundationElement {
+export class TableCell<
+    TCellData extends TableRecord = TableRecord
+> extends FoundationElement {
     // TODO: This should be replaced with an instance of TableCellState<TCellData>
     @observable
-    public data: TableFieldValue;
+    public data?: TableCellState<TCellData>;
+
+    @observable
+    public cellTemplate?: ViewTemplate;
+
+    @observable
+    public cellStyles?: ElementStyles;
+
+    private customCellView: HTMLView | undefined = undefined;
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.updateView();
+    }
+
+    private updateView(): void {
+        const newCellView = this.customCellView === undefined;
+        if (newCellView) {
+            this.customCellView = this.cellTemplate!.create(this);
+            if (this.cellStyles) {
+                this.$fastController.addStyles(this.cellStyles);
+            }
+        }
+
+        this.customCellView?.bind(this.data, defaultExecutionContext);
```

Comment:
For reviewers, this is the reason we are creating the cell content in code-behind instead of in the template.